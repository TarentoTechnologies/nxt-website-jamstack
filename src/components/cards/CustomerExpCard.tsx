import React from "react";

import { h4400 } from "../../styles/style-guide/Typography.module.css";
import { cxCard } from "./CustomerExpCard.module.css";

interface CustomerExpCardProps {
  id: any;
  title?: string;
  logo?: any;
  isSecondary?: boolean;
  description?: string;
  link?: string;
  isLink?: boolean;
}

export const CustomerExpCard = ({
  id,
  title,
  description,
  logo,
  isSecondary,
  link,
  isLink,
}: CustomerExpCardProps) => {
  if (isLink) {
    return (
      <a
        href={link != null ? link : ""}
        target="_blank"
        rel="noreferrer"
        id={id}
        data-testid="customer-exp-card"
      >
        <div className="col-lg-3 col-md-6 col-sm-12 marginBtm1">
          <div className="" data-aos="fade-up">
            <div className={cxCard}>
              <img
                className="centerImg"
                src={logo}
                alt={title}
                data-aos="fade-up"
              />
            </div>
          </div>
          <div className="">
            <h4 className={`${h4400} mt-4`} data-aos="fade-up">
              {title}
            </h4>
          </div>
          <div className="mt-4 mb-4">
            <p data-aos="fade-up">{description}</p>
          </div>
        </div>
      </a>
    );
  } else {
    return (
      <div
        className="col-lg-3 col-md-6 col-sm-12 marginBtm1"
        id={id}
        data-testid="customer-exp-card"
      >
        <div className="" data-aos="fade-up">
          <div className={cxCard}>
            <img
              className="centerImg"
              src={logo}
              alt={title}
              data-aos="fade-up"
            />
          </div>
        </div>
        <div className="">
          <h4 className={`${h4400} mt-4`} data-aos="fade-up">
            {title}
          </h4>
        </div>
        <div className="mt-4 mb-4">
          <p data-aos="fade-up">{description}</p>
        </div>
      </div>
    );
  }
};
