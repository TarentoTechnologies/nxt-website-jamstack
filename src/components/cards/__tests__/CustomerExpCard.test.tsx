import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { CustomerExpCard } from "../CustomerExpCard";

describe("Test if the CustomerExpCard component is working as expected", () => {
  it("<CustomerExpCard /> matches snapshot", () => {
    const component = render(
      <CustomerExpCard
        id="b72407f9-f344-56a7-b959-8c53650a273d"
        title="Intuitive Experience"
        description="Make the experience as intuitive as possible. Avoid chances for errors, confusion. Ensure that your users have sufficient self help materials when they want to know more. Capture telemetry to understand user behaviour anonymously."
        logo="https://strapi-stage-nxt.tarento.com/uploads/experience_285006c250.svg"
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<CustomerExpCard /> renders without exploding", () => {
    const component = render(
      <CustomerExpCard
        id="b72407f9-f344-56a7-b959-8c53650a273d"
        title="Intuitive Experience"
        description="Make the experience as intuitive as possible. Avoid chances for errors, confusion. Ensure that your users have sufficient self help materials when they want to know more. Capture telemetry to understand user behaviour anonymously."
        logo="https://strapi-stage-nxt.tarento.com/uploads/experience_285006c250.svg"
      />
    );
    expect(component.getByTestId("customer-exp-card")).toBeInTheDocument();
  });
});
