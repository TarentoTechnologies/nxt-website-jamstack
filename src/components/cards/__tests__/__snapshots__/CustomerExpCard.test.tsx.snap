// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the CustomerExpCard component is working as expected <CustomerExpCard /> matches snapshot 1`] = `
<div>
  <div
    class="col-lg-3 col-md-6 col-sm-12 marginBtm1"
    data-testid="customer-exp-card"
    id="b72407f9-f344-56a7-b959-8c53650a273d"
  >
    <div
      class=""
      data-aos="fade-up"
    >
      <div
        class="cxCard"
      >
        <img
          alt="Intuitive Experience"
          class="centerImg"
          data-aos="fade-up"
          src="https://strapi-stage-nxt.tarento.com/uploads/experience_285006c250.svg"
        />
      </div>
    </div>
    <div
      class=""
    >
      <h4
        class="h4400 mt-4"
        data-aos="fade-up"
      >
        Intuitive Experience
      </h4>
    </div>
    <div
      class="mt-4 mb-4"
    >
      <p
        data-aos="fade-up"
      >
        Make the experience as intuitive as possible. Avoid chances for errors, confusion. Ensure that your users have sufficient self help materials when they want to know more. Capture telemetry to understand user behaviour anonymously.
      </p>
    </div>
  </div>
</div>
`;
